_includes/live-demos/mentions/index.js

```diff
@@ -1,241 +1,165 @@
-      name: name,
-      fullName: fullName,
-      description: description,
-      image: image
+/* Script to import faker.js for generating random data for demonstration purposes */
+tinymce.ScriptLoader.loadScripts(['https://cdn.jsdelivr.net/npm/faker@5/dist/faker.min.js'], function() {
+
+  /*
+  ** This is to simulate requesting information from a server.
+  **
+  ** It has 2 functions:
+  ** fetchUsers() - returns a complete list of users' ids and names.
+  ** fetchUser(id) - returns the full information about a single user id.
+  **
+  ** Both of these functions have a slight delay to simulate a server request.
+  */
+  var fakeServer = (function () {
+    /* Use TinyMCE's Promise shim */
+    var Promise = tinymce.util.Promise;
+
+    /* Some user profile images for our fake server (original source: unsplash) */
+    var images = [
+      'Abdullah_Hadley', 'Abriella_Bond', 'Addilynn_Dodge', 'Adolfo_Hess', 'Alejandra_Stallings', 'Alfredo_Schafer', 'Aliah_Pitts', 'Amilia_Luna', 'Andi_Lane', 'Angelina_Winn', 'Arden_Dean', 'Ariyanna_Hicks', 'Asiya_Wolff', 'Brantlee_Adair', 'Carys_Metz', 'Daniela_Dewitt', 'Della_Case', 'Dianna_Smiley', 'Eliana_Stout', 'Elliana_Palacios', 'Fischer_Garland', 'Glen_Rouse', 'Grace_Gross', 'Heath_Atwood', 'Jakoby_Roman', 'Judy_Sewell', 'Kaine_Hudson', 'Kathryn_Mcgee', 'Kayley_Dwyer', 'Korbyn_Colon', 'Lana_Steiner', 'Loren_Spears', 'Lourdes_Browning', 'Makinley_Oneill', 'Mariana_Dickey', 'Miyah_Myles', 'Moira_Baxter', 'Muhammed_Sizemore', 'Natali_Craig', 'Nevaeh_Cates', 'Oscar_Khan', 'Rodrigo_Hawkins', 'Ryu_Duke', 'Tripp_Mckay', 'Vivianna_Kiser', 'Yamilet_Booker', 'Yarely_Barr', 'Zachary_Albright', 'Zahir_Mays', 'Zechariah_Burrell'
+    ];
+
+    /* Create an array of 200 random names using faker.js */
+    var userNames = [];
+    for (var i = 0; i < 200; i++) {
+      userNames.push(faker.name.findName());
     };
-  }).forEach(function(user) {
-    userDb[user.id] = user;
-  });
 
-  /* This represents getting the complete list of users from the server with the details required for the mentions "profile" item */
-  var fetchUsers = function() {
-    return new Promise(function(resolve, _reject) {
-      /* simulate a server delay */
-      setTimeout(function() {
-        var users = Object.keys(userDb).map(function(id) {
-          return {
-            id: id,
-            name: userDb[id].name,
-            image: userDb[id].image,
-            description: userDb[id].description
-          };
-        });
-        resolve(users);
-      }, 500);
+    /* This represents a database of users on the server */
+    var userDb = {};
+    userNames.map(function (fullName) {
+      var id = fullName.toLowerCase().replace(/ /g, '');
+      return {
+        id: id,
+        name: fullName,
+        fullName: fullName,
+        description: faker.name.jobTitle(),
+        image: '{{ site.baseurl }}/images/unsplash/uifaces-unsplash-portrait-' + images[Math.floor(images.length * Math.random())] + '.jpg'
```

Comment:
We can't do much about this duplicated code with docs being the way they are, can we

Reply:
no, not really. It's mainly duped because of the `scriptloader` bit, otherwise we could reuse rather than dupe

Reply:
tragic